### ISSUE TITLE
additionalProperties not working when set to a JSON object and not to a boolean

### ISSUE THREAD

**Issue by giacomomarchioro** [koxudaxi/datamodel-code-generator#696]
Thanks for this wonderful tool!
**Describe the bug**
From the [json schema specification ](http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties) if I understand well, a JSON object can be used in `additionalProperties` like below:

**To Reproduce**

Example schema:
```json

{
    "$id": "https://example.com/person.schema.json",
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "title": "my JSON of list of string",
    "type": "object",
    "additionalProperties" : {
            "type": "array",
            "items": { "type": "string"}
        }
}

```

Used commandline:
```
datamodel-codegen --input jsonschema.json --input-file-type jsonschema  --output skeleton.py
```

**Expected behavior**
Now I would expect that only `"test":["test"]` or `"testb":["testb","testc"]` are valid while instead it seems that any dict is valid:

```
import skeleton 
t = skeleton.JSONofListOfString()
t.validate({"asdf":[1,2,3,4]}) 
t.validate({"asdf":1}) 
```

**Version:**
 - OS: MacOS 10.15.7
 - Python version: 3.8.3
 - datamodel-code-generator version: 0.11.8

**Comment by koxudaxi** [comment]
Hi @giacomomarchioro,
I'm sorry for my late reply.

I tried your example 
```python

from __future__ import annotations

from pydantic import BaseModel, Extra


class MyJsonOfListOfString(BaseModel):
    pass

    class Config:
        extra = Extra.allow
```

What output do you expect?
I don't have the idea that I can realize your request in pydantic.

**Comment by koxudaxi** [comment]
@giacomomarchioro 
I'm sorry for my too-late reply.  
I have released the PR as `0.15.0`
Thank you very much!!

**Comment by giacomomarchioro** [comment]
Thanks a lot @koxudaxi  !
